chimerapy/chimera_copy.py

```diff
@@ -0,0 +1,938 @@
+    r = thresh_rad
+    w = np.where((xgir - center[0]) ** 2 + (ygir - center[1]) ** 2 <= thresh_rad**2)
+    cir[w] = 1.0
+    cand = can * cir
+    return r, w, cir, cand
+
+
+r, w, cir, cand = misid(cand, circ, xgrid, ygrid, (s[1] / 2.0) - 100)
+
+
+def on_off(cir: np.array, can: np.array):
+    """
+    Seperates on-disk and off-limb coronal holes
+
+    Parameters
+    ----------
+    cir: 'np.array'
+    can: 'np.array'
+
+    Returns
+    -------
+    'np.array'
+
+    """
+    cir[:] = 0
+    r = (rs.value / dattoarc) - 10
+    inside = np.where((xgrid - center[0]) ** 2 + (ygrid - center[1]) ** 2 <= r**2)
+    cir[inside] = 1.0
+    r = (rs.value / dattoarc) + 40
+    outside = np.where((xgrid - center[0]) ** 2 + (ygrid - center[1]) ** 2 >= r**2)
+    cir[outside] = 1.0
+    can = can * cir
+    plt.figure()
+    plt.imshow(cand, cmap="viridis")
+    plt.show
+    return can
+
+
+cand = on_off(circ, cand)
+
+
+def contour_data(cand: np.array):
+    """
+    Contours the identified datapoints
+
+    Parameters
+    ----------
+    cand: 'np.array'
+
+    Returns
+    -------
+    cand: 'np.array'
+    cont: 'tuple'
+    heir: 'np.array'
+
+    """
+    cand = np.array(cand, dtype=np.uint8)
+    cont, heir = cv2.findContours(cand, cv2.RETR_TREE, cv2.CHAIN_APPROX_SIMPLE)
+    # I think cont might be the x-y coordinates in pixels?
+    return cand, cont, heir
+
+
+cand, cont, heir = contour_data(cand)
+
+
+def sort(cont: tuple):
+    """
+    Sorts the contours by size
+
+    Parameters
+    ----------
+    cont: 'tuple'
+
+    Returns
+    -------
+    reord: 'list'
+    tmp: 'list'
+    cont: 'list'
+    sizes: 'list'
+
+    """
+    sizes = []
+    for i in range(len(cont)):
+        sizes = np.append(sizes, len(cont[i]))
+    reord = sizes.ravel().argsort()[::-1]
+    tmp = list(cont)
+    for i in range(len(cont)):
+        tmp[i] = cont[reord[i]]
+    cont = list(tmp)
+    return cont, sizes, reord, tmp
+
+
+cont, sizes, reord, tmp = sort(cont)
+
+
+# =====cycles through contours=========
+
+
+def extent(map: sunpy.map.Map, cont: tuple):
+    """
+    Finds coronal hole extent in latitude and longitude
+
+    Parameters
+    ----------
+    map: 'sunpy.map.Map'
+    cont: 'tuple'
+
+    Returns
+    -------
+    maxxlon: 'astropy.coordinates.angles.core.Longitude'
+    minxlon: 'astropy.coordinates.angles.core.Longitude'
+    centlat: 'astropy.coordinates.angles.core.Latitude'
+    centlon: 'astropy.coordinates.angles.core.Longitude'
+
+    """
+
+    coord_hpc = map.world2pix(cont)
+    maxlat = coord_hpc.transform_to(HeliographicStonyhurst).lat.max()
+    maxlon = coord_hpc.transform_to(HeliographicStonyhurst).lon.max()
+    minlat = coord_hpc.transform_to(HeliographicStonyhurst).lat.min()
+    minlon = coord_hpc.transform_to(HeliographicStonyhurst).lat.min()
+
+    # =====CH centroid in lat/lon=======
+
+    centlat = hg.lat[int(ypos), int(xpos)]
+    centlon = hg.lon[int(ypos), int(xpos)]
```

Comment:
Where do `ypos`, `xpos` come from there they should be function arguments if they are used?

Reply:
They were inputs in my last version - not sure why I deleted them but I'm including them again now

Reply:
Just looked a little more closely, they are the x and y positions of the center point which are used later in the for loop. I assumed we'd change that so they weren't calculated manually which I think is why I got rid of them.